### ISSUE TITLE
Add proper h2 elements for accessibility

### ISSUE THREAD

**Issue by nickvergessen** [nextcloud/spreed#7103]
The interface needs proper h2 elements, e.g.:
- [x] Conversation list
- [x] [Conversation title] as it shows in header
- [x] some title for the right sidebar (which is ideally not the same as the Conversation title since it would duplicate, could be "Conversation details")
- [x] Requires a new release with
    - https://github.com/nextcloud/nextcloud-vue/pull/2865
    - https://github.com/nextcloud/nextcloud-vue/pull/2845
- [ ] https://github.com/nextcloud/spreed/pull/7623 ❓ 

Extracted from https://github.com/nextcloud/spreed/issues/6801

**Comment by jancborchardt** [comment]
> Or do we simply hide it from the screen so it's just visible for screenreaders?

Yep, this – since it doesn't really make sense visually to have a heading for the navigation.

**Comment by nikola-gladovic** [comment]
@nickvergessen 
1. conversation list: can not edit the conversation title of each record, because it's in ListItem component which come from nextcloud library
3. right sidebar: I don't see any title can update to h2. Please take a look and let me know

**Comment by nickvergessen** [comment]
1. It's not about the conversation entries themself but about the sections. E.g. When you start to search:

![Bildschirmfoto von 2022-04-27 08-27-52](https://user-images.githubusercontent.com/213943/165455182-e084cbcf-f760-4307-856a-5583415a5b73.png)

Each of the "blue" sections is a "caption", but it's not an h2 so it's not recognizable for screenreaders => I guess this needs adjusting in the vue library:
https://github.com/nextcloud/nextcloud-vue/blob/master/src/components/AppNavigationCaption/AppNavigationCaption.vue#L15-L17

And when you are not searching we should also have the "conversations" AppNavigationCaption but hidden visual so screenreaders still have it.

**Comment by nickvergessen** [comment]
In the right sidebar we currently still have the conversation name (which is duplicated).
It should be called "Conversation details" there, but before that we need to move renaming of a conversation to the conversation settings modal.

You can do that in 1 PR if you want to

**Comment by nickvergessen** [comment]
Right sidebar fixed by https://github.com/nextcloud/nextcloud-vue/pull/2845

**Comment by nickvergessen** [comment]
Conversation list/filters fixed with https://github.com/nextcloud/nextcloud-vue/pull/2865

**Comment by nickvergessen** [comment]
@AndyScherzinger ????

**Comment by AndyScherzinger** [comment]
@nickvergessen all todos are marked as done, so I thought, let's close it

**Comment by nickvergessen** [comment]
There is still a PR linked to it...
https://github.com/nextcloud/spreed/pull/7623

**Comment by AndyScherzinger** [comment]
ah, you referenced another one
whoops, sry
